src/testing/slim_check/sampleable.lean

```diff
@@ -118,6 +118,13 @@ instance sampleable_nat : sampleable ℕ :=
                           coe <$> choose_any (fin $ succ sz),
   shrink := lazy_list.of_list ∘ nat.shrink }
 
+instance sampleable_fin {n} [fact $ 0 < n] : sampleable (fin n) :=
```

Comment:
Probably a question for the larger architecture, but: why is nonemptiness required here? If we view `sampleable` as the possible domains for things which can be `testable`, then empty domains should be okay, as long as we can say that it is empty. That probably means putting things in an option monad.

Reply:
`gen` does not have an error mode but if you want to test empty domains, you could write a `testable` instance that just returns `success`. If you were to add an error mode for `gen`, it would be more useful to interpret it as `gave_up`.